nautobot/core/models/__init__.py

```diff
@@ -1,11 +1,58 @@
 import uuid
+from urllib.parse import quote_plus
 
+from django.contrib.contenttypes.models import ContentType
 from django.db import models
+from natural_keys import NaturalKeyModel, NaturalKeyModelManager
+from natural_keys.models import extract_nested_key
 
 from nautobot.core.models.querysets import RestrictedQuerySet
 
 
-class BaseModel(models.Model):
+class BaseManager(NaturalKeyModelManager):
```

Comment:
I'm thinking it's probably best to have this live in `nautobot.core.models.managers` just to preempt any potential import issues.